pkg/dns/serviceresolver.go

```diff
@@ -115,13 +119,14 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 				if fallback, ok := b.fallback(prefix, exact); ok {
 					return b.Records(fallback+b.base, exact)
 				}
+				return nil, errNoSuchName
 			}
-			return nil, err
+			return nil, errNoSuchName
```

Comment:
It's been long enough that I've forgotten.  I don't remember whether SRVFAIL is the right behavior.